modules/candidate_parameters/ajax/formHandler.php

```diff
@@ -0,0 +1,397 @@
+<?php
+/**
+ * Candidate parameters form handler
+ *
+ * Handles form update actions received from a front-end ajax call
+ *
+ * PHP Version 5
+ *
+ * @category Loris
+ * @package  Media
+ * @author   Stella L. <slee.mcin@gmail.com>
+ * @license  Loris license
+ * @link     https://github.com/aces/Loris-Trunk
+ */
+if (isset($_POST['tab'])) {
+    $tab = $_POST['tab'];
+    if ($tab == "candidateInfo") {
+        editCandInfoFields();
+    } else if ($tab == "probandInfo") {
+        editProbandInfoFields();
+    } else if ($tab == "familyInfo") {
+        editFamilyInfoFields();
+    } else if ($tab == "deleteFamilyMember") {
+        deleteFamilyMember();
+    } else if ($tab == "participantStatus") {
+        editParticipantStatusFields();
+    } else if ($tab == "consentStatus") {
+        editConsentStatusFields();
+    } else {
+        header("HTTP/1.1 404 Not Found");
+        exit;
+    }
+}
+
+$db   =& Database::singleton();
+$user =& User::singleton();
+
+/**
+ * Handles the updating of Candidate Info
+ *
+ * @throws DatabaseException
+ *
+ * @return void
+ */
+function editCandInfoFields()
+{
+    if (!$this->user->hasPermission('candidate_parameter_edit')) {
```

Comment:
You can technically declare:
```
$db   =& Database::singleton();
$user =& User::singleton();
```
on line 17 right after `$tab`, and then pass references to all methods.

Otherwise use `global` keyword (probably not best practice)